File: aoc2023/test/aoc2023/day01_test.clj
(ns aoc2023.day01-test
  (:require [clojure.test :refer :all]
            [clojure.java.io]
            [aoc2023.day01 :as day01]))

(def sample-input (slurp (clojure.java.io/resource "day01-sample.txt")))
(def sample-input2 (slurp (clojure.java.io/resource "day01-sample2.txt")))
(def sample-calibration (day01/load-calibration sample-input))

(deftest test-part1
  (testing "Day 1 Part 1"
    (is (= 142 (day01/part1 sample-calibration)))
  )
)

(deftest test-part2
  (testing "Day 1 Part 2"
    (is (= 281 (day01/part2 sample-input2)))
  )
)

File: 2024/day01/day01_test.go
package day01

import (
	"fmt"
    "testing"
    )

var sample = "sample.txt"

func TestPart1(t *testing.T) {
  result := Part1(sample)
  if result != 11 {
    fmt.Println(result)
  	t.Fail()
  }
}

func TestPart2(t *testing.T) {
  result := Part2(sample)
  if result != 31 {
    fmt.Println(result)
  	t.Fail()
  }
}
